test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/TagHelperDescriptorFactoryTest.cs

```diff
@@ -13,6 +17,161 @@ public class TagHelperDescriptorFactoryTest
         private static readonly string AssemblyName =
             typeof(TagHelperDescriptorFactoryTest).GetTypeInfo().Assembly.GetName().Name;
 
+        public static TheoryData AttributeTargetData
+        {
+            get
+            {
+                var attributes = Enumerable.Empty<TagHelperAttributeDescriptor>();
+
+                // tagHelperType, expectedDescriptors
+                return new TheoryData<Type, IEnumerable<TagHelperDescriptor>>
+                {
+                    {
+                        typeof(AttributeTargetingTagHelper),
+                        new[]
+                        {
+                            new TagHelperDescriptor(
+                                "*",
```

Comment:
nit: please use the `TagHelperDescriptorProvider.CatchAllDescriptorTarget` field for clarity